src/renderer/routes/test_sequencer_panel/components/CloudPanel.tsx

```diff
@@ -13,18 +13,73 @@ import {
 import { Button } from "@/renderer/components/ui/button";
 import { useAppStore } from "@/renderer/stores/app";
 import { useShallow } from "zustand/react/shallow";
-import { testSequenceExportCloud } from "@/renderer/routes/test_sequencer_panel/models/models";
-import { getCloudProjects, getEnvironmentVariables } from "@/renderer/lib/api";
+import {
+  Project,
+  Station,
+  Unit,
+  getCloudProjects,
+  getCloudStations,
+  getCloudUnits,
+  getEnvironmentVariables,
+} from "@/renderer/lib/api";
 import { toastQueryError } from "@/renderer/utils/report-error";
 import { useQuery, useQueryClient } from "@tanstack/react-query";
 import { Spinner } from "@/renderer/components/ui/spinner";
+import { TestSequenceContainer } from "@/renderer/types/test-sequencer";
+import { Badge } from "@/renderer/components/ui/badge";
+import { Checkbox } from "@/renderer/components/ui/checkbox";
+import useWithPermission from "@/renderer/hooks/useWithPermission";
+import { toast } from "sonner";
+import { getGlobalStatus } from "./DesignBar";
+import { useSequencerStore } from "@/renderer/stores/sequencer";
+import { useAuth } from "@/renderer/context/auth.context";
+import { Autocomplete } from "@/renderer/components/ui/autocomplete";
 
 export function CloudPanel() {
   const queryClient = useQueryClient();
-  const [hardwareId, setHardwareId] = useState("");
+  const { isAdmin } = useWithPermission();
+  const [lotNumber, setLotNumber] = useState("");
+  const [description, setDescription] = useState("N/A");
   const [projectId, setProjectId] = useState("");
-  const { tree, setIsLocked } = useTestSequencerState();
-  const { tSSendJsonMessage } = useTestSequencerWS();
+  const [partNumber, setPartNumber] = useState("");
+  const [partVarId, setPartVarId] = useState("");
+  const [serialNumbers, setSerialNumbers] = useState<string[]>([]);
+  const [units, setUnits] = useState<Map<string, Unit>>(new Map());
```

Comment:
it's not worth it to use `Map` with immutable state, just use a `Record` so you get spread syntax

Reply:
`Map` is the right element here since I don't know all the possible key, the facts that it's a state is a problem tho.
- Converted to a simple map variable

Reply:
`Map` and `Record` do basically the same thing though, it would just be `Record<string, Unit>` which can easily be updated immutably